backend/src/main/kotlin/com/game/main/GameWebSocket.kt

```diff
@@ -129,6 +129,9 @@ class GameWebSocket {
                         broadcastNextTurnMessage(game)
                     }
                 }
+                if(game.checkGameOver()){
```

Comment:
This should probably be a branch within the above nested structure of if-elses. The check should happen at the end of a round, rather than every time a player chooses hit or miss.